src/Compression/TaskToApm.cs

```diff
@@ -60,9 +60,15 @@ public static TResult End<TResult>(IAsyncResult asyncResult)
             if (asyncResult is TaskAsyncResult twar && twar._task is Task<TResult> task)
             {
                 return task.GetAwaiter().GetResult();
+            } else if (asyncResult is null) {
+                ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (asyncResult));
+            } else {
+                ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (twar._task));
```

Comment:
I thought these were better to give us more detailed exceptions, but the compiler does not recognize these `ObjCRuntime.ThrowHelper.ThrowArgumentNullException`s as ending the method so I kept the old one at the end of the method...